cartconditions/Coupon.php

```diff
@@ -89,7 +89,7 @@ public function onLoad()
     public function beforeApply()
     {
         $couponModel = $this->getModel();
-        if (!$couponModel OR $couponModel->is_limited_to_cart_item)
+        if (!$couponModel)// OR $couponModel->is_limited_to_cart_item)
```

Comment:
`beforeApply` is only called when the condition is being applied to the entire cart, so here we want to make sure we are not applying a cart item condition on the entire cart.

Reply:
If we don't call this then none of the validation rules are called on the coupon... it relates quite closely to https://github.com/tastyigniter/ti-ext-coupons/pull/19 - see my reply there.